lua/packer.lua

```diff
@@ -663,71 +663,84 @@ end
+  async(function()
+    if not donot_move_plugins then
+      local update = require_and_configure 'update'
+      local plugin_utils = require_and_configure 'plugin_utils'
+      local fs_state = await(plugin_utils.get_fs_state(plugins))
+      await(a.main)
+      update.fix_plugin_types(plugins, vim.tbl_keys(fs_state.missing), {}, fs_state)
+    end
+    local args = parse_args(raw_args)
+    local output_path = args.output_path or config.compile_path
+    local output_lua = vim.fn.fnamemodify(output_path, ':e') == 'lua'
+    local should_profile = args.profile
+    -- the user might explicitly choose for this value to be false in which case
+    -- an or operator will not work
+    if should_profile == nil then
+      should_profile = config.profile.enable
+    end
+    refresh_configs(plugins)
+    -- NOTE: we copy the plugins table so the in memory value is not mutated during compilation
+    local compiled_loader = compile(vim.deepcopy(plugins), output_lua, should_profile)
+    output_path = vim.fn.expand(output_path)
+    vim.fn.mkdir(vim.fn.fnamemodify(output_path, ':h'), 'p')
+    local output_file = io.open(output_path, 'w')
+    output_file:write(compiled_loader)
+    output_file:close()
+    if config.auto_reload_compiled then
+      local configs_to_run = {}
+      if _G.packer_plugins ~= nil then
+        for plugin_name, plugin_info in pairs(_G.packer_plugins) do
+          if plugin_info.loaded and plugin_info.config and plugins[plugin_name] and plugins[plugin_name].cmd then
+            configs_to_run[plugin_name] = plugin_info.config
+          end
         end
       end
-    end
 
-    vim.cmd('source ' .. output_path)
-    for plugin_name, plugin_config in pairs(configs_to_run) do
-      for _, config_line in ipairs(plugin_config) do
-        local success, err = pcall(loadstring(config_line))
-        if not success then
-          vim.notify('Error running config for ' .. plugin_name .. ': ' .. vim.inspect(err), vim.log.levels.ERROR, {})
+      vim.cmd('source ' .. output_path)
+      for plugin_name, plugin_config in pairs(configs_to_run) do
+        for _, config_line in ipairs(plugin_config) do
+          local success, err = pcall(loadstring(config_line))
+          if not success then
+            vim.notify('Error running config for ' .. plugin_name .. ': ' .. vim.inspect(err), vim.log.levels.ERROR, {})
+          end
         end
       end
     end
-  end
-  log.info 'Finished compiling lazy-loaders!'
-  packer.on_compile_done()
-
-  -- TODO: remove this after migration period (written 2021/06/28)
-  local old_output_path
-  if output_lua then
-    old_output_path = vim.fn.fnamemodify(output_path, ':r') .. '.vim'
-  else
-    old_output_path = vim.fn.fnamemodify(output_path, ':r') .. '.lua'
-  end
+    log.info 'Finished compiling lazy-loaders!'
+    print 'PackerCompile: Done'
+    packer.on_compile_done()
+
+    -- TODO: remove this after migration period (written 2021/06/28)
```

Comment:
Doesn't need to be part of this PR, but we're out of the migration period and should remove this shim.

Reply:
Done